pkg/search/postgres/store_cache.go

```diff
@@ -294,13 +295,28 @@ func (c *cachedStore[T, PT]) GetMany(ctx context.Context, identifiers []string)
 }
 
 // WalkByQuery iterates over all the objects scoped by the query applies the closure.
-func (c *cachedStore[T, PT]) WalkByQuery(ctx context.Context, q *v1.Query, fn func(obj PT) error) error {
-	if q == nil || protocompat.Equal(q, search.EmptyQuery()) {
-		c.cacheLock.RLock()
-		defer c.cacheLock.RUnlock()
-		return c.walkCacheNoLock(ctx, fn)
```

Comment:
I'd keep that optimisation: I don't see the point of going to the underlying store if we know all items should be returned.